app/registries/foreman/plugin.rb

```diff
@@ -492,21 +492,23 @@ def smart_proxy_reference(hash)
       @smart_proxy_references << ProxyReferenceRegistry.new_reference(hash)
     end
 
+    def register_renderer_variable_loader(loader_name)
+      @renderer_variable_loaders << loader_name
+    end
+
     private
 
+    def extend_template_helpers_by_module(mod)
+      Foreman::Renderer::Scope::Base.class_eval do
+        send(:include, mod.to_s.constantize)
```

Comment:
I believe REX should just define it's own scope then and not use the plugin interfgace. Actually, the REX PR does that right now. ;-)

Reply:
Right, for extending just REX that makes sense. I think it might be useful if plugins want to extend only report templates scope and not anything else. But not necessary for this PR. I'm fine with adding that later in case it would be needed.